cirq/google/calibration/test_data/xeb_results.textproto

```diff
@@ -0,0 +1,274 @@
+timestamp_ms: 1615418725184
+metrics {
+  name: "initial_fidelities_by_depth"
+  targets: "0_0_3_qubit_a"
+  targets: "0_0_3_qubit_b"
```

Comment:
I don't think this is the intended use of this metrics proto.
The intended use is that the targets should be the qubit targets with the values.
So, i think this should be
metrics{
   name: 'initial_fidelities_depth_3'
   targets: '0_0'
   targets: '0_1'
   values {
    double_val: 0.9932020253010514
  }
}
And have one of those for each pair.  Alternatively, you could have multiple values for each depth if it would be clear which depths go to which values.

This would avoid the magic naming problem (for the most part) of having very specific target names and value ordering.

Reply:
yeah, I was following the floquet calibration schema which is quite odd. There would still be some data hidden in the name (i.e. the depth).

One other thing I must point out: in the cirq implementation, I explicitly avoided using the qubit pair (i.e. the two grid qubits) as a key for a particular fidelity, angle, or anything. This is because you could submit a multi-layer (more than one moment) request for calibration where the same pair appears in several layers. There would be a collision if we tried to serialize the results using just the qubits as targets. Following the floquet structure, the actual `xxRequest` `xxResponse` classes only support doing single-layer calibrations at a time (albeit submitted as a batch) so this may not be the biggest issue. But it may cause problems down the line if we start doing multilayer calibrations.

I think this could be remedied by adding the `pair_i, layer_i` indices to the targets. This takes away some of the beauty of just having the qubits be the target like how I imagine Metric proto was designed for

Reply:
I've prototyped a new format that looks like:

<details>

```
timestamp_ms: 1616131925329
metrics {
  name: "initial_fidelities_depth_3"
  targets: "layer_0"
  targets: "pair_0"
  targets: "0_0"
  targets: "0_1"
  values {
    double_val: 1.0012187701926922
  }
}
metrics {
  name: "initial_fidelities_depth_23"
  targets: "layer_0"
  targets: "pair_0"
  targets: "0_0"
  targets: "0_1"
  values {
    double_val: 0.9999390106596308
  }
}
metrics {
  name: "initial_fidelities_depth_43"
  targets: "layer_0"
  targets: "pair_0"
  targets: "0_0"
  targets: "0_1"
  values {
    double_val: 1.0026270943879811
  }
}
metrics {
  name: "initial_fidelities_depth_63"
  targets: "layer_0"
  targets: "pair_0"
  targets: "0_0"
  targets: "0_1"
  values {
    double_val: 0.9944886126006032
  }
}
metrics {
  name: "initial_fidelities_depth_83"
  targets: "layer_0"
  targets: "pair_0"
  targets: "0_0"
  targets: "0_1"
  values {
    double_val: 1.0061950635412118
  }
}
metrics {
  name: "initial_fidelities_depth_3"
  targets: "layer_0"
  targets: "pair_1"
  targets: "0_2"
  targets: "0_3"
  values {
    double_val: 0.999143029103658
  }
}
metrics {
  name: "initial_fidelities_depth_23"
  targets: "layer_0"
  targets: "pair_1"
  targets: "0_2"
  targets: "0_3"
  values {
    double_val: 0.9960486039906495
  }
}
metrics {
  name: "initial_fidelities_depth_43"
  targets: "layer_0"
  targets: "pair_1"
  targets: "0_2"
  targets: "0_3"
  values {
    double_val: 0.9902634563212708
  }
}
metrics {
  name: "initial_fidelities_depth_63"
  targets: "layer_0"
  targets: "pair_1"
  targets: "0_2"
  targets: "0_3"
  values {
    double_val: 0.995208414628978
  }
}
metrics {
  name: "initial_fidelities_depth_83"
  targets: "layer_0"
  targets: "pair_1"
  targets: "0_2"
  targets: "0_3"
  values {
    double_val: 1.0041419375484237
  }
}
metrics {
  name: "final_fidelities_depth_3"
  targets: "layer_0"
  targets: "pair_0"
  targets: "0_0"
  targets: "0_1"
  values {
    double_val: 1.0129396387456167
  }
}
metrics {
  name: "final_fidelities_depth_23"
  targets: "layer_0"
  targets: "pair_0"
  targets: "0_0"
  targets: "0_1"
  values {
    double_val: 1.0496164580926466
  }
}
metrics {
  name: "final_fidelities_depth_43"
  targets: "layer_0"
  targets: "pair_0"
  targets: "0_0"
  targets: "0_1"
  values {
    double_val: 0.98448873586971
  }
}
metrics {
  name: "final_fidelities_depth_63"
  targets: "layer_0"
  targets: "pair_0"
  targets: "0_0"
  targets: "0_1"
  values {
    double_val: 0.9466170806619504
  }
}
metrics {
  name: "final_fidelities_depth_83"
  targets: "layer_0"
  targets: "pair_0"
  targets: "0_0"
  targets: "0_1"
  values {
    double_val: 1.1924106455204362
  }
}
metrics {
  name: "final_fidelities_depth_3"
  targets: "layer_0"
  targets: "pair_1"
  targets: "0_2"
  targets: "0_3"
  values {
    double_val: 0.9999230056806732
  }
}
metrics {
  name: "final_fidelities_depth_23"
  targets: "layer_0"
  targets: "pair_1"
  targets: "0_2"
  targets: "0_3"
  values {
    double_val: 0.9971401591030289
  }
}
metrics {
  name: "final_fidelities_depth_43"
  targets: "layer_0"
  targets: "pair_1"
  targets: "0_2"
  targets: "0_3"
  values {
    double_val: 0.9901261409944282
  }
}
metrics {
  name: "final_fidelities_depth_63"
  targets: "layer_0"
  targets: "pair_1"
  targets: "0_2"
  targets: "0_3"
  values {
    double_val: 0.9934397763794856
  }
}
metrics {
  name: "final_fidelities_depth_83"
  targets: "layer_0"
  targets: "pair_1"
  targets: "0_2"
  targets: "0_3"
  values {
    double_val: 1.0043742401838296
  }
}
metrics {
  name: "characterized_angles_phi"
  targets: "0_0"
  targets: "0_1"
  targets: "pair_0"
  values {
    double_val: 0.05625000000000001
  }
}
metrics {
  name: "characterized_angles_phi"
  targets: "0_2"
  targets: "0_3"
  targets: "pair_1"
  values {
    double_val: 0.003125
  }
}
```

</details>

Reply:
What does `pair_i` stands for? Why just `layer_i` wouldn't be suffice?

Reply:
Yeah, this is looking better.  I think we should get rid of the layer_i and pair_i in the targets though.  I am not sure why we need them.

One optional suggestion (I am not sure if it is better or worse), is to have a metric called "depths" and then have the values be [3,23,43,83] and then make the metrics for the fidelities multi-valued.  

Also, if there is changes needed to the results creation on pyle-side with regards to limitations there, we should go ahead and change it there. It was supposed to make editing the proto easier, so we should add convenience functions when needed.

Reply:
`pair_i` is technically redundant. For `layer_i`: you can imagine a request to calibrate the same pair in two different spacial contexts, i.e. the pair appears in two different layers.

The current Request objects only lets you do one layer at a time, but the web api lets you send a full circuit. The backend handler will happily do multiple moments (unless we restrict it).